fml/raster_thread_merger.cc

```diff
@@ -128,32 +120,35 @@ void RasterThreadMerger::WaitUntilMerged() {
     return;
   }
   FML_CHECK(IsOnPlatformThread());
-  std::unique_lock<std::mutex> lock(lease_term_mutex_);
+  std::unique_lock<std::mutex> lock(mutex_);
   merged_condition_.wait(lock, [&] { return IsMergedUnSafe(); });
 }
 
 RasterThreadStatus RasterThreadMerger::DecrementLease() {
   if (TaskQueuesAreSame()) {
     return RasterThreadStatus::kRemainsMerged;
   }
-  std::unique_lock<std::mutex> lock(lease_term_mutex_);
+  std::scoped_lock lock(mutex_);
   if (!IsMergedUnSafe()) {
     return RasterThreadStatus::kRemainsUnmerged;
   }
   if (!IsEnabledUnSafe()) {
     return RasterThreadStatus::kRemainsMerged;
   }
-  FML_DCHECK(lease_term_ > 0)
-      << "lease_term should always be positive when merged.";
-  lease_term_--;
-  if (lease_term_ == 0) {
-    // |UnMergeNow| is going to acquire the lock again.
-    lock.unlock();
-    UnMergeNow();
+  bool unmerged_after_decrement = shared_merger_impl_->DecrementLease();
```

Comment:
This too, while threadsafe, what are the ramifications of 2 thread mergers decrementing?  Would we be decrementing the lease twice as fast as intended?  @iskakaushik

Reply:
Yes I think it's a big problem..
I used the demo for test and printed some log in some case:
Case 1: two webview: MergedWithLease(10) and  lease_term_ decrease from 10 to 9, and because there's a if (HasPlatformLayers()) and ExtendLease() was called, it cames back to 10 again.
Case 2: show two webviews and removed two webviews: MergedWithLease(10) and first merger let lease_term_ decrease from 10 to 9, the second changed it from 9 to 8.  etc.. Finally lease came to zero and UnmergeNow() was called. Yes the speed is twice as fast as one merger
Case 3: show two webviews and remove one webview(one webview still existed): MergedWithLease(10) and first merger let lease_term_ decrease from 10 to 9, the second changed it from 9 to 8, the first rasterizer called ExtendLease(10), 8 changed to 10.
So in case 3, if there's 10 rasterizers(10 merger instances) and only 1 flutter engine shows platform view, It will be a bug...

Reply:
I will find a better way to solve this bug.😄

Reply:
@gaaclarke Solved this multi-merger decrementing bug by:
https://github.com/flutter/engine/pull/27662/commits/1d48d1574d7e146fa29a742c732ea97b90146031